gateway/api.go

```diff
@@ -375,12 +373,16 @@ func (gw *Gateway) basicAuthHashAlgo() string {
 	config := gw.GetConfig()
 
 	// Use `basic_auth_hash_key_function` if set;
-	if config.BasicAuthHashKeyFunction != "" {
-		return config.BasicAuthHashKeyFunction
+	algo := config.BasicAuthHashKeyFunction
+
+	// If hash function name is empty/invalid
+	if ok := user.IsHashType(algo); !ok {
+		// set default basic auth hash to bcrypt
+		return "bcrypt"
```

Comment:
shall we use the const `HashBCrypt` here?

Reply:
I'd leave it for now - no strong argument pro/contra, would be `string(user.HashBCrypt)`, as outside of the user package `string` is used, even tho a `user.HashType` type exists; The responsibility of password hashing could be it's own package on itself (internal/hash), rather than having a storage/user/gateway fragmented implementation.

Reply:
Added it. :)